adapter_files/precice.cpp

```diff
@@ -524,14 +528,19 @@ double Precice::advance(double computedTimestepLength) {
       double displacementDeltas_su2[vertexSize[i]][nDim]; /*--- displacementDeltas will be stored such, before
                                                              converting to simple array ---*/
 
-      if (!dataTypeDisplacement) {
-        displacementDeltas = new double[vertexSize[i] * nDim];
-        solverInterface.readBlockVectorData(displDeltaID[indexMarkerWetMappingLocalToGlobal[i]], vertexSize[i],
-                                            vertexIDs[i], displacementDeltas);
-      } else {
-        displacements = new double[vertexSize[i] * nDim];
-        solverInterface.readBlockVectorData(displDeltaID[indexMarkerWetMappingLocalToGlobal[i]], vertexSize[i],
-                                            vertexIDs[i], displacements);
+      switch (readDataType) {
+        case ReadDataType::DisplacementDelta: {
+          displacementDeltas = new double[vertexSize[i] * nDim];
+          solverInterface.readBlockVectorData(displDeltaID[indexMarkerWetMappingLocalToGlobal[i]], vertexSize[i],
+                                              vertexIDs[i], displacementDeltas);
+          break;
+        }
+        case ReadDataType::Displacement: {
+          displacements = new double[vertexSize[i] * nDim];
+          solverInterface.readBlockVectorData(displDeltaID[indexMarkerWetMappingLocalToGlobal[i]], vertexSize[i],
+                                              vertexIDs[i], displacements);
+          break;
+        }
```

Comment:
You could add a `default: assertion(false)`.

Reply:
Done